From 8d3c667788090b877f2d7504f1ec2115ea05bb3e Mon Sep 17 00:00:00 2001
From: Dan McDonald <danmcd@omniti.com>
Date: Mon, 3 Oct 2016 19:31:12 +0000
Subject: [PATCH] OS-5696 lint error in contract_process_fork() OS-5697 lxbrand
 references readlinkat() with wrong type Reviewed by: Patrick Mooney
 <patrick.mooney@joyent.com> Reviewed by: Jerry Jelinek
 <jerry.jelinek@joyent.com> Approved by: Jerry Jelinek
 <jerry.jelinek@joyent.com>

---
 usr/src/uts/common/brand/lx/syscall/lx_link.c | 2 +-
 usr/src/uts/common/contract/process.c         | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/usr/src/uts/common/brand/lx/syscall/lx_link.c b/usr/src/uts/common/brand/lx/syscall/lx_link.c
index e8deaf54f1..4ebf491d23 100644
--- a/usr/src/uts/common/brand/lx/syscall/lx_link.c
+++ b/usr/src/uts/common/brand/lx/syscall/lx_link.c
@@ -30,7 +30,7 @@ extern int unlinkat(int, char *, int);
 /* From uts/common/syscall/symlink.c */
 extern int symlinkat(char *, int, char *);
 /* From uts/common/syscall/readlink.c */
-extern int readlinkat(int, char *, char *, size_t);
+extern ssize_t readlinkat(int, char *, char *, size_t);
 
 static long
 lx_link_common(int ffd, char *from, int tfd, char *to, int flags)
diff --git a/usr/src/uts/common/contract/process.c b/usr/src/uts/common/contract/process.c
index e92f87eecf..e46cbd3abf 100644
--- a/usr/src/uts/common/contract/process.c
+++ b/usr/src/uts/common/contract/process.c
@@ -1078,7 +1078,7 @@ contract_process_fork(ctmpl_process_t *rtmpl, proc_t *cp, proc_t *pp,
 	 * abandon it now.
 	 */
 	if (ctp->conp_params & CT_PR_KEEP_EXEC) {
-		contract_abandon(ct, pp, 1);
+		(void) contract_abandon(ct, pp, 1);
 	}
 
 	return (ctp);
-- 
2.21.0

